src/screens/OtherProfile/index.js

```diff
@@ -70,42 +62,47 @@ const OtherProfile = () => {
   const specificIssueRef = React.useRef();
   const flatListRef = React.useRef();
 
-  const [dataMain, setDataMain] = React.useState({});
-  const [, setDataMainBio] = React.useState('');
   const [user_id, setUserId] = React.useState('');
   const [username, setUsername] = React.useState('');
   const [other_id, setOtherId] = React.useState('');
   const [isShowButton, setIsShowButton] = React.useState(false);
   const [opacity, setOpacity] = React.useState(0);
   const [reason, setReason] = React.useState([]);
   const [yourselfId] = React.useState('');
-  const [blockStatus, setBlockStatus] = React.useState({
-    blocked: false,
-    blocker: false
-  });
   const [loadingBlocking, setLoadingBlocking] = React.useState(false);
-  const [postOffset, setPostOffset] = React.useState(0);
 
   const headerHeightRef = React.useRef(0);
   const interactionManagerRef = React.useRef(null);
-  const [otherProfileFeeds, dispatchOtherProfile] = React.useContext(Context).otherProfileFeed;
+  const [mainFeeds, dispatchOtherProfile] = React.useContext(Context).otherProfileFeed;
   const [profile] = React.useContext(Context).profile;
-  const [mainFeeds, dispatch] = React.useContext(Context).feeds;
-  const [loading, setLoading] = React.useState(false);
-  const [initLoading, setInitLoading] = React.useState(true);
+  const {timer, viewPostTimeIndex} = mainFeeds;
+  const [, dispatch] = React.useContext(Context).feeds;
   const [isAnonimity, setIsAnonimity] = React.useState(false);
   const {params} = route;
-  const {feeds} = otherProfileFeeds;
-  const {timer, viewPostTimeIndex} = mainFeeds;
   const [loadingGenerateAnon, setLoadingGenerateAnon] = React.useState(false);
   const [anonProfile, setAnonProfile] = React.useState();
-  const {mappingColorFeed} = useCoreFeed();
+  const {createSignChat} = useCreateChat();
+
+  const {
+    feeds,
+    otherProfileData: dataMain,
+    isBlocked,
+    isBlocking,
+    isLoading,
+    isProfileFetching,
+    postOffset,
+
+    refetchFeeds,
+    refetchBlockStatus,
+    refetchOtherProfile,
+    setOtherProfileData: setDataMain
+  } = useOtherProfileScreenHooks(params?.data?.other_id, params?.data?.username);
+
   const isSignedMessageEnabled = dataMain.isSignedMessageEnabled ?? true;
   const isAnonimityEnabled = dataMain.isAnonMessageEnabled && isSignedMessageEnabled;
-  const {createSignChat} = useCreateChat();
 
   const {onWillSendViewPostTime} = useViewPostTimeHook(dispatch, timer, viewPostTimeIndex);
-  const {fetchNextFeeds} = useFeedPreloadHook(feeds?.length, () => getOtherFeeds(postOffset));
+  const {fetchNextFeeds} = useFeedPreloadHook(feeds?.length, () => refetchFeeds(postOffset));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code has removed the offline mode handling and block status checking which were present in the old code. If these functionalities are not handled elsewhere, this could lead to issues when the device is offline or when a user's block status needs to be checked.



<!-- This is an auto-generated comment by OSS CodeRabbit -->